tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,248 @@ fn generate_synthetic(
                     ),
                 )
                 .into();
-                first_transaction.append_instruction(mint);
+                first_transaction.push_instruction(mint);
             }
         }
     }
 
     Ok(genesis)
 }
+
+/// Use `Kagami` to sign genesis block.
+#[derive(ClapArgs, Clone, Debug)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+#[command(group = ArgGroup::new("format").required(false))]
+pub struct SignArgs {
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: crypto::AlgorithmArg,
+    /// Private key (in string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
```

Comment:
> (in string format)

Which one? A JSON with `algorithm` and `payload` fields?

Reply:
As there's a field for algorithm I thought of using hex string for private key and multihash for public. I've added `hex` part

Reply:
You mean, SCALE-encoded hex?
It is not a usual format which users use with Iroha for private keys. Unfortunately, we always separate `algorithm` and `payload` somehow :(

This might help: https://github.com/hyperledger/iroha/issues/4412

Reply:
Please replace `String` with a type that shows _what_ this string is, which shows how it parses it.